modules/gis/src/lib/geojson-to-binary.js

```diff
@@ -0,0 +1,205 @@
+export function featuresToBinary(features, options = {}) {
+  const firstPassData = firstPass(features);
+  return secondPass(features, firstPassData, options);
+}
+
+function firstPass(features) {
+  // Counts the number of _positions_, so [x, y, z] counts as one
+  let pointPositions = 0;
+  let linePositions = 0;
+  let linePaths = 0;
+  let polygonPositions = 0;
+  let polygonObjects = 0;
+  let polygonRings = 0;
+  let maxCoordLength = 2;
+
+  features.forEach(feature => {
+    const geometry = feature.geometry;
+    switch (geometry.type) {
+      case 'Point':
+        pointPositions++;
+        if (geometry.coordinates.length === 3) maxCoordLength = 3;
+        break;
+      case 'MultiPoint':
+        pointPositions += geometry.coordinates.length;
+        geometry.coordinates.forEach(point => {
+          if (point.length === 3) maxCoordLength = 3;
+        });
+        break;
+      case 'LineString':
+        linePositions += geometry.coordinates.length;
+        linePaths++;
+
+        geometry.coordinates.forEach(coord => {
+          if (coord.length === 3) maxCoordLength = 3;
+        });
+        break;
+      case 'MultiLineString':
+        geometry.coordinates.forEach(line => {
+          linePositions += line.length;
+          linePaths++;
+
+          line.coordinates.forEach(coord => {
+            if (coord.length === 3) maxCoordLength = 3;
+          });
+        });
+        break;
+      case 'Polygon':
+        polygonObjects++;
+        polygonRings += geometry.coordinates.length;
+        polygonPositions += geometry.coordinates.flat(1).length;
+
+        geometry.coordinates.flat().forEach(coord => {
+          if (coord.length === 3) maxCoordLength = 3;
+        });
+        break;
+      case 'MultiPolygon':
+        geometry.coordinates.forEach(polygon => {
+          polygonObjects++;
+          polygonRings += polygon.length;
+          polygonPositions += polygon.flat(1).length;
+
+          geometry.coordinates.flat().forEach(coord => {
+            if (coord.length === 3) maxCoordLength = 3;
+          });
+        });
+        break;
+      default:
+        throw new Error(`Unsupported geometry type: ${geometry.type}`);
+    }
+  });
+
+  return {
+    pointPositions,
+    linePositions,
+    linePaths,
+    coordLength: maxCoordLength,
+    polygonPositions,
+    polygonObjects,
+    polygonRings
+  };
+}
+
+function secondPass(features, firstPassData, options = {}) {
+  const {
+    pointPositions,
+    linePositions,
+    linePaths,
+    coordLength,
+    polygonPositions,
+    polygonObjects,
+    polygonRings
+  } = firstPassData;
+  const {PositionDataType = Float32Array} = options;
+  const points = {
+    positions: new PositionDataType(pointPositions * coordLength),
```

Comment:
For next PR: should probably wrap these in the  "accessor objects" that we use across luma/deck/loaders to provide metadata for binary arrays to prep them for use in WebGL frameworks. At the very least, provide `size`

```
{
  points: {
    positions: {values: new ..., size: coordLength}, 
    objectIds: {values: new ..., size: 1}
  },
  lines: {
```

Reply:
Yes I agree this is necessary. It doesn't hurt to do it now, since we're waiting for 2.1 to be released anyways.

Reply:
From the [deck.gl performance guide](https://deck.gl/#/documentation/developer-guide/performance-optimization?section=supply-attributes-directly) it seems to be `value` without the `s`?